cpp/src/DataStorm/SessionI.cpp

```diff
@@ -678,16 +670,15 @@ SessionI::retry(NodePrx node, exception_ptr exception)
                 << " (ms) for peer to reconnect";
         }
 
+        // Schedule a timer to remove the session if the peer doesn't reconnect.
         _retryTask = make_shared<IceInternal::InlineTimerTask>([self = shared_from_this()] { self->remove(); });
```

Comment:
Here `SessionI::remove` would call retry again, and we schedule a new call to remove.

As far as I see we have and endless loop of calling retry to print this message and scheduling the remove call that just calls retry.